ingestion/tests/unit/topology/pipeline/test_dagster.py

```diff
@@ -0,0 +1,197 @@
+
+
+EXPECTED_PIPELINE_STATUS = [
+    OMetaPipelineStatus(
+        pipeline_fqn="dagster_source.a10f6d82-4fc6-4c90-ba04-bb773c8fbb0f",
+        pipeline_status=PipelineStatus(
+            executionStatus=StatusType.Pending.value,
+            taskStatus=[
+                TaskStatus(
+                    name="a10f6d82-4fc6-4c90-ba04-bb773c8fbb0f",
+                    executionStatus=StatusType.Pending.value,
+                    startTime=1659616627124,
+                    endTime=1659616635858,
+                    logLink=f"{MOCK_LOG_URL}/status",
+                )
+            ],
+            executionDate=1659616635858,
+        ),
+    ),
+    OMetaPipelineStatus(
+        pipeline_fqn="dagster_source.a10f6d82-4fc6-4c90-ba04-bb773c8fbb0f",
+        pipeline_status=PipelineStatus(
+            executionStatus=StatusType.Successful.value,
+            taskStatus=[
+                TaskStatus(
+                    name="a10f6d82-4fc6-4c90-ba04-bb773c8fbb0f",
+                    executionStatus=StatusType.Successful.value,
+                    startTime=1655393914,
+                    endTime=1655394054,
+                    logLink=f"{MOCK_LOG_URL}/status",
+                )
+            ],
+            executionDate=1655393914,
+        ),
+    ),
+]
+
+
+EXPECTED_CREATED_PIPELINES = CreatePipelineRequest(
+    name="a6ebb16c-505f-446d-8642-171c3320ccef",
+    displayName="do_it_all_with_default_configs",
+    description="do_it_all_with_default_config",
+    pipelineUrl=MOCK_CONNECTION_URI_PATH,
+    tasks=[
+        Task(
+            name="do_it_all_with_default_config",
+            displayName="",
+            description="",
+            taskUrl="",
+        )
+    ],
+    service=EntityReference(
+        id="85811038-099a-11ed-861d-0242ac120002", type="pipelineService"
+    ),
+)
+
+MOCK_PIPELINE_SERVICE = PipelineService(
+    id="86ff3c40-7c51-4ff5-9727-738cead28d9a",
+    name="dagster_source_test",
+    connection=PipelineConnection(),
+    serviceType=PipelineServiceType.Dagster,
+)
+
+
+MOCK_PIPELINE = Pipeline(
+    id="a58b1856-729c-493b-bc87-6d2269b43ec0",
+    name="do_it_all_with_default_config",
+    fullyQualifiedName="dagster_source.do_it_all_with_default_config",
+    displayName="do_it_all_with_default_config",
+    description="",
+    pipelineUrl=MOCK_CONNECTION_URI_PATH,
+    tasks=[
+        Task(
+            name="a58b1856-729c-493b-bc87-6d2269b43ec0",
+            displayName="do_it_all_with_default_config",
+            description="",
+            taskUrl="",
+        )
+    ],
+    service=EntityReference(
+        id="85811038-099a-11ed-861d-0242ac120002", type="pipelineService"
+    ),
+)
+
+
+class DagsterUnitTest(TestCase):
+    @patch("metadata.ingestion.source.pipeline.pipeline_service.test_connection")
+    @patch("metadata.ingestion.source.pipeline.dagster.DagsterClient")
+    def __init__(self, methodName, test_connection) -> None:
```

Comment:
I think we are using only one patch here, we can remove one which is not required